Code/src/lin.cpp

```diff
@@ -31,40 +31,34 @@ Lin::Lin(LIN_SERIAL& ser,uint8_t TxPin)
 {
 }
 
+// define bit period without () so that integer calculations can be more accurate
+#define TBIT 1000000UL/serialSpd
 void Lin::begin(int speed)
 {
   serialSpd = speed;
   serial.begin(serialSpd);
-  serialOn  = 1;
 
-  unsigned long int Tbit = 100000/serialSpd;  // Not quite in uSec, I'm saving an extra 10 to change a 1.4 (40%) to 14 below...
-  unsigned long int nominalFrameTime = ((34*Tbit)+90*Tbit);  // 90 = 10*max # payload bytes + checksum (9). 
-  timeout = LIN_TIMEOUT_IN_FRAMES * 14 * nominalFrameTime;  // 14 is the specced addtl 40% space above normal*10 -- the extra 10 is just pulled out of the 1000000 needed to convert to uSec (so that there are no decimal #s).
```

Comment:
Original code here has multiple mistakes, and the comments dont help.
This is a large reduction of the timeout value and works fine.

Reply:
timeout was 17360us. now 6458us.